### ISSUE TITLE
Invalid explicit SpyreTensorLayout causes SIGFPE during CPU-to-Spyre copy

### ISSUE THREAD

**Issue by cyang49** [torch-spyre/torch-spyre#3401]
### What happened?

An invalid explicit `SpyreTensorLayout` is accepted by the constructor, but copying a tensor to Spyre with that layout terminates the Python process with `SIGFPE`.

The layout metadata is invalid, so rejecting it is expected. The bug is that it causes an unrecoverable process crash instead of a validation exception.

#### Suggestions

- Describe the crashing layout in the issue as “malformed or unsupported under the current SpyreTensorLayout contract,” rather than simply “invalid.”
- Request early validation with a clear exception. If arbitrary axis arrangements are intended to be supported, then the issue should instead request DCI support for that layout.

### What did you expect to happen?

The copy should raise a Python-visible `RuntimeError` or `TORCH_CHECK` explaining that the device sizes and stride map are incompatible with the host tensor.

Invalid user-provided layout metadata should not terminate the process.

### How can we reproduce it?

```python
import torch
import torch_spyre

from torch.spyre import DataFormats, SpyreTensorLayout

x = torch.rand((1, 4, 4096, 128), dtype=torch.float16)

invalid_layout = SpyreTensorLayout(
    device_size=[1, 4, 128, 64, 64],
    stride_map=[2097152, 524288, 1, 8192, 128],
    device_dtype=DataFormats.SEN169_FP16,
)

y = x.to(device="spyre", device_layout=invalid_layout)
torch.spyre.synchronize()
```

Observed result:

```text
Signal Received: 8 (Floating point exception)
...
torch_spyre/_C.so(spyre::get_device_stride_infos(...))
...
Process exited with status 255
```

For comparison, this corrected layout for the same tensor succeeds:

```python
valid_layout = SpyreTensorLayout(
    device_size=[1, 4, 64, 128, 64],
    stride_map=[2097152, 524288, 8192, 1, 128],
    device_dtype=DataFormats.SEN169_FP16,
)
```

### Any environmental details we need to know?

No

### Anything else we need to know?

The likely failure is in `get_device_stride_infos()` in `torch_spyre/csrc/spyre_mem.cpp`.

For the invalid layout, this calculation can produce `tile_stride == 0`:

```cpp
const int64_t tile_stride = host_strides[tile_index] / host_stride;
const int64_t remaining_elements = current_elements / tile_stride;
```

The division occurs before any check that `tile_stride` is positive. A similar risk may exist for `next_stride` later in the remainder path.

This issue does not require accepting the invalid layout—the desired behavior is a controlled validation error.

### Relevant log output

```Shell

```

**Comment by postmath** [comment]
Related: #3604 is a second `SIGFPE` in `spyre::get_device_stride_infos`, but on a **different divisor at a different line** — `host_size % elements_before` (line 266, `elements_before == 0`) rather than `current_elements / tile_stride` (line 271, `tile_stride == 0`). Both were reproduced under gdb to confirm the distinction.

Flagging it here because a narrow fix for this issue — validating the public `SpyreTensorLayout` constructor, or guarding only `tile_stride` — would not fix #3604. There the layout is generated internally by the compiler (`_rescale_stl_for_dtype` floors a size-1 dim to 0), so no user-supplied layout is involved and constructor validation never sees it.

The "early validation with a clear exception" suggested here would cover both, provided it validates every divisor in `get_device_stride_infos` rather than just the one on this path.
